### PR TITLE
get test suite passing on Windows

### PR DESCRIPTION
**Pull request by SethTisue**
this was manually tested (by ssh'ing to a Cygwin prompt on
jenkins-worker-windows-publish), since
https://github.com/scala/scala-jenkins-infra/issues/36 isn't done yet

review by @adriaanm, @som-snytt, @retronym, @martijnhoekstra

### CODE REVIEW

**Comment by SethTisue** [comment]
oh, I should have mentioned, if anyone is tempted to actually run this on Windows themselves, I wouldn't currently suggest that you bother. from Cygwin (we're not worrying about from cmd.exe anymore) some parts of the ant build require `set -o igncr`, others require you _not_ have that... it's messy. I'll need to straighten that out before scala/scala-jenkins-infra#36 can be closed.

File: src/partest-extras/scala/tools/partest/IcodeComparison.scala [hunk 1]
```diff
@@ -48,8 +48,13 @@ abstract class IcodeComparison extends DirectTest {
...
 
-    try     icodeFiles sortBy (_.name) flatMap (f => f.lines.toList)
-    finally icodeFiles foreach (f => f.delete())
+    // Some methods in scala.reflect.io.File leak an InputStream, leaving the underlying file open.
+    // Windows won't delete an open file, but we must ensure the files get deleted, since the logic
+    // here depends on it (collectIcode will be called multiple times, and we can't allow crosstalk
+    // between calls).  So we are careful to use `slurp` which does call `close`, and careful to
+    // check that `delete` returns true indicating successful deletion.
+    try     icodeFiles sortBy (_.name) flatMap (f => f.slurp().lines.toList)
+    finally icodeFiles foreach (f => require(f.delete()))
```
Comment (som-snytt) [inline src/partest-extras/scala/tools/partest/IcodeComparison.scala:57]: As a matter of fact, I recall that the fix to slurp was because I ran tests on windows and hit the same issue.

https://github.com/scala/scala/blob/v2.11.7/src/reflect/scala/reflect/io/Streamable.scala#L112

Li Haoyi just complained about the behavior of the other methods (he was using Source I think) and I checked that indeed there are leakages, which is too bad.

OTOH, at the Java office, I just had to verify that when using try-with-res, you have to close wrapped resources or be very sure about the behavior. End of this article points out that not all close methods guarantee to close decorated resources.

http://www.oracle.com/technetwork/articles/java/trywithresources-401775.html

So basically, Damn you all to hell! [PAN to reveal STATUE OF LIBERTY half-buried in the sand.]

**Comment by lrytz** [comment]
LGTM `\r\n`

File: src/partest-extras/scala/tools/partest/IcodeComparison.scala (hunk 1 above)
Comment (som-snytt) [inline src/partest-extras/scala/tools/partest/IcodeComparison.scala:57]: `require(icodeFiles forall (_.delete))`. Or `try ff flatMap (f => f.slurp.lines.toList.ensuring(f.delete))` except exceptions. Could use a macro `enforcing` that wrapped (some) tree in the try.
